src/lib.rs

```diff
@@ -171,108 +170,146 @@ use std::sync::atomic::{AtomicIsize, Ordering, ATOMIC_ISIZE_INIT};
 
 use apm::Listener;
 use bson::Bson;
-use common::{ReadPreference, ReadMode, WriteConcern};
+use common::{ReadPreference, WriteConcern};
 use connstring::ConnectionString;
 use db::{Database, ThreadedDatabase};
 use error::Error::ResponseError;
 use pool::PooledStream;
-use stream::StreamConnector;
-use topology::{Topology, TopologyDescription, TopologyType, DEFAULT_HEARTBEAT_FREQUENCY_MS,
-DEFAULT_LOCAL_THRESHOLD_MS, DEFAULT_SERVER_SELECTION_TIMEOUT_MS};
+use stream::ConnectMethod;
+use topology::{Topology, TopologyDescription, DEFAULT_HEARTBEAT_FREQUENCY_MS,
+               DEFAULT_LOCAL_THRESHOLD_MS, DEFAULT_SERVER_SELECTION_TIMEOUT_MS};
 use topology::server::Server;
 
 /// Interfaces with a MongoDB server or replica set.
 pub struct ClientInner {
-    /// Indicates how a server should be selected for read operations.
```

Comment:
As mentioned in the summary, I removed the read preference and write concern options from the client.